src/OutputFormatter/JUnitOutputFormatter.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace SensioLabs\Deptrac\OutputFormatter;
+
+use Fhaculty\Graph\Vertex;
+use SensioLabs\AstRunner\AstMap;
+use SensioLabs\Deptrac\ClassNameLayerResolverInterface;
+use SensioLabs\Deptrac\Dependency\Result;
+use SensioLabs\Deptrac\DependencyContext;
+use SensioLabs\Deptrac\DependencyResult\DependencyInterface;
+use SensioLabs\Deptrac\DependencyResult\InheritDependency;
+use SensioLabs\Deptrac\RulesetEngine\RulesetViolation;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Jan Schädlich <janschaedlich@sensiolabs.de>
+ */
+final class JUnitOutputFormatter implements OutputFormatterInterface
+{
+    private static $argument_dump_xml = 'dump-xml';
+
+    public function getName(): string
+    {
+        return 'junit';
+    }
+
+    /**
+     * @return OutputFormatterOption[]
+     */
+    public function configureOptions(): array
+    {
+        return [
+            OutputFormatterOption::newValueOption(static::$argument_dump_xml, 'path to a dumped xml file', ''),
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     * @throws \Exception
+     */
+    public function finish(
+        DependencyContext $dependencyContext,
+        OutputInterface $output,
+        OutputFormatterInput $outputFormatterInput
+    ) {
+
+        $xml = $this->createXml($dependencyContext);
+
+        if ($dumpXmlPath = $outputFormatterInput->getOption(static::$argument_dump_xml) ?: './junit-report.xml') {
+            file_put_contents($dumpXmlPath, $xml);
+            $output->writeln('<info>JUnit Report dumped to ' . realpath($dumpXmlPath) . '</info>');
+        }
+    }
+
+    /**
+     * @param DependencyContext $dependencyContext
+     * @return string
+     * @throws \Exception
+     */
+    private function createXml(DependencyContext $dependencyContext)
+    {
+        if (!class_exists("\DomDocument")) {
```

Comment:
I think you have misunderstood the comment. The line should look like this:

```
if (!class_exists(\DomDocument::class)) {
```

This does not throw any errors if the class does not exist. The class name is resolved at compile time. ;-)

Reply:
compile time? ^^

Reply:
PHP use an JIT-Compiler (Just in Time Compiler) so PHP files are run in two stages `Compile Time` & `Run Time`: 

PHP Source Code -(Compile Time)-> Bytecode -(Run Time)-> Code Execution

In the compiler step, the `::class` expression is converted to a string. See:

https://3v4l.org/fHY7h/vld#output

And here's an example with a nonexistent class:

https://3v4l.org/f0hVj/vld#output